code/datums/status_effects/neutral.dm

```diff
@@ -620,21 +620,46 @@
 	desc = "Waaater... Fuck this WATER!!"
 	icon_state = "shower_regen_catgirl"
 
+/atom/movable/screen/alert/status_effect/shower_regen/bloody_like
+	name = "Washing"
+	desc = "Mhhhmmmm... the crimson red drops of life. How delightful."
+	icon_state = "shower_regen_blood_happy"
+
+/atom/movable/screen/alert/status_effect/shower_regen/bloody_dislike
+	name = "Washing"
+	desc = "Is that... blood? What the fuck!"
+	icon_state = "shower_regen_blood_bad"
+
 /datum/status_effect/shower_regen
 	id = "shower_regen"
 	duration = -1
 	status_type = STATUS_EFFECT_UNIQUE
 	alert_type = /atom/movable/screen/alert/status_effect/shower_regen
 	/// How many heals from washing.
 	var/stamina_heal_per_tick = 4
+	/// Is this the shower raining blood?
+	var/bloody_shower = FALSE
+
+/datum/status_effect/shower_regen/on_creation(mob/living/new_owner, _bloody_shower = FALSE)
+	bloody_shower = _bloody_shower
+	return ..()
 
 /datum/status_effect/shower_regen/on_apply()
 	. = ..()
-	if(HAS_TRAIT(owner, TRAIT_WATER_HATER) && !HAS_TRAIT(owner, TRAIT_WATER_ADAPTATION))
+
+	if(bloody_shower)
+		if(HAS_TRAIT(owner, TRAIT_MORBID) || HAS_TRAIT(owner, TRAIT_EVIL) || (owner.mob_biotypes & MOB_UNDEAD))
+			alert_type = /atom/movable/screen/alert/status_effect/shower_regen/bloody_like
+		else
+			alert_type  = /atom/movable/screen/alert/status_effect/shower_regen/bloody_dislike
+	else if(HAS_TRAIT(owner, TRAIT_WATER_HATER) && !HAS_TRAIT(owner, TRAIT_WATER_ADAPTATION))
 		alert_type = /atom/movable/screen/alert/status_effect/shower_regen/hater
 
 /datum/status_effect/shower_regen/tick(seconds_between_ticks)
 	. = ..()
+	if(bloody_shower)
+		return
```

Comment:
Now blood arbitrarily doesn't interact the same as all the other TOUCH chems used in a shower. I suppose just removing the interaction for blood technically works, but this is kind of a shitty solution.

Reply:
The old behavior was that if you had ANY chemical (even if it was NOT water) the regen effects happen and you got a positive mood boost for showering in something like oil or acid.

What should the mood boost be for showering in 3u water, 3u oil, 3u acid, 1u blood? What if the blood is infected? Which alert should be used? What about regen effects? etc. This was all stuff I considered. I ultimately decided to just use a boolean solution. Does the water have blood or not? Doesn't matter if it's 99% or 0.001%.

I'm also not sure where you are seeing the other chems getting bypassed? It only applies to the mood/regen effects. All the other TOUCH chem interactions should get be getting triggered.

Reply:
I'm not a maint, so you are welcome to ignore me if you want since I'm asking a lot out of you rather unfairly. But it doesn't quite sit right with me to just switch it off if there is ever blood in the shower. I would expect there to just be a degen effect if there is more blood than other reagents in the shower, unless the person is happy with a bloody shower. Namely, it's only a bloody shower if there is majority blood in the reagents overall.

It might require a deeper amount of refactoring, but I think it's entirely possible. Probably by getting the volume of blood and comparing it to the volume of total reagents, and if it is above half, it's a bloody shower.